lib/util/exec/types.ts

```diff
@@ -45,7 +45,7 @@ export interface ExecOptions {
   extraEnv?: Opt<ExtraEnv>;
   docker?: Opt<DockerOptions>;
   toolConstraints?: Opt<ToolConstraint[]>;
-  preCommands?: Opt<string[]>;
+  preCommands?: Opt<string>[];
```

Comment:
nope, we don't accept null or undefined strings here. should be filtered from caller